src/PBE/ExecutableContainer.cs

```diff
@@ -21,100 +21,114 @@ public class ExecutableContainer
+                return fsPath;
             }
 
-            // Prio4 - Starndard Installations-Verzeichnis:
+            // Prio4 - Standard Installations-Verzeichnis:
             // - Framework Studio\x.y
-            string searchDir = Path.Combine(
-                Environment.GetEnvironmentVariable("ProgramFiles"), 
-                "enventa Group", 
-                "Framework Studio", 
-                new Version(fsVersion).ToString(2));
-            if (File.Exists(Path.Combine(searchDir, "FSConsole.exe")))
+            dir = Path.Combine(Environment.GetEnvironmentVariable("ProgramFiles"), "enventa Group", "Framework Studio");
+            if (TryGetFSVersionPath(dir, fsVer, out fsPath))
             {
-                this.FsVerdict.TryAdd(fsVersion, searchDir);
-                return searchDir;
+                this.FsVerdict.TryAdd(fsVersion, fsPath);
+                return fsPath;
             }
 
-            // Prio5 - altes Starndard Installations-Verzeichnis in folgender Priorisierung:
+            // Prio5 - altes Standard Installations-Verzeichnis in folgender Priorisierung:
             // - Framework Studio x.y.12
             // - Framework Studio x.y.2
             // - Framework Studio x.y
             // - Framework Studio x.y Beta2
             // - Framework Studio x.y Beta
-            string searchDirOld = Path.Combine(Environment.GetEnvironmentVariable("ProgramFiles"), "Framework Systems");
-            if (Directory.Exists(searchDirOld))
+            dir = Path.Combine(Environment.GetEnvironmentVariable("ProgramFiles"), "Framework Systems");
+            if (TryGetFSVersionPath(dir, fsVer, out fsPath))
             {
-                Version maxVersion = null;
-                int? maxBeta = null;
-                string maxPath = null;
-
-                var version = new Version(fsVersion);
-                string searchPattern = "Framework Studio " + version.ToString(2) + "*";
-                foreach (var path in Directory.EnumerateDirectories(searchDirOld, searchPattern)
-                    .Where(path => File.Exists(Path.Combine(path, "FSConsole.exe"))))
+                this.FsVerdict.TryAdd(fsVersion, fsPath);
+                return fsPath;
+            }
+
+            throw new ApplicationException("No installation found for Framework Studio " + fsVersion + ".");
+        }
+
+        private bool TryGetFSVersionPath(string basePath, FSVersion fsVersion, out string fsPath)
+        {
+            fsPath = string.Empty;
+
+            if (!Directory.Exists(basePath))
+                return false;
+
+            string pattern = $@"^(Framework Studio )?{Regex.Escape(fsVersion.Version.ToString(2))}(((\.\d*)*)?)?";
+
+            var fsDirs = Directory.EnumerateDirectories(basePath)
+                            .Where(d => Regex.IsMatch(Path.GetFileName(d), pattern, RegexOptions.IgnoreCase))
+                            .Where(d => File.Exists(Path.Combine(d, "FSConsole.exe")));
```

Comment:
Die Enumertation wird mehrfach ausgeführt - einmal die Any()-Prüfung und später nochmal.
Ohne `ToArray()` (oder eine andere Persistierung) wird mehrmals die Platte gescannt.

```suggestion
            var fsDirs = Directory.EnumerateDirectories(basePath)
                            .Where(d => Regex.IsMatch(Path.GetFileName(d), pattern, RegexOptions.IgnoreCase))
                            .Where(d => File.Exists(Path.Combine(d, "FSConsole.exe")))
                            .ToArray();
```